src/Native/System.Native/pal_io.cpp

```diff
@@ -149,7 +149,8 @@ static void ConvertFileStatus(const struct stat_& src, FileStatus* dst)
 extern "C" int32_t Stat(const char* path, FileStatus* output)
 {
     struct stat_ result;
-    int ret = stat_(path, &result);
+    int ret;
+    while (CheckIo(ret = stat_(path, &result)));
```

Comment:
I don't know how reliable/credible this site is, but https://www.mkssoftware.com/docs/man3/stat.3.asp lists EINTR for stat, lstat and fstat, and the their 64 variations.